maskrcnn_benchmark/modeling/poolers.py

```diff
@@ -108,14 +120,21 @@ def forward(self, x, boxes):
         output_size = self.output_size[0]
 
         dtype, device = x[0].dtype, x[0].device
-        result = torch.zeros(
-            (num_rois, num_channels, output_size, output_size),
-            dtype=dtype,
-            device=device,
-        )
-        for level, (per_level_feature, pooler) in enumerate(zip(x, self.poolers)):
-            idx_in_level = torch.nonzero(levels == level).squeeze(1)
-            rois_per_level = rois[idx_in_level]
-            result[idx_in_level] = pooler(per_level_feature, rois_per_level)
-
+        if 0:  # facilitate scripting
+            result = torch.zeros(
+                (num_rois, num_channels, output_size, output_size),
+                dtype=dtype,
+                device=device,
+            )
+            for level, (per_level_feature, pooler) in enumerate(zip(x, self.poolers)):
+                idx_in_level = torch.nonzero(levels == level).squeeze(1)
+                rois_per_level = rois[idx_in_level]
+                result[idx_in_level] = pooler(per_level_feature, rois_per_level)
+        else:
+            unmerged_results = []
+            for level, (per_level_feature, pooler) in enumerate(zip(x, self.poolers)):
+                idx_in_level = torch.nonzero(levels == level).squeeze(1)
+                rois_per_level = rois[idx_in_level]
+                unmerged_results.append(pooler(per_level_feature, rois_per_level))
+            result = merge_levels(levels, unmerged_results)
```

Comment:
I haven't benchmarked, but my intuition would be that this isn't terribly bad for performance.
I don't think we can easily pass the ModuleList to a script function, so we cannot put the entire loop there.
We could make try to make the pooling for loop a separate ScriptModule.

Reply:
This should be fine performance-wise.
I have a question: with the changes from https://github.com/pytorch/pytorch/pull/13093, can we use `scatter_` from python in tracing or not yet?

Reply:
Awesome! It doesn't work in tracing, but in script it does! We can trade the custom op merge_levels for a torch.jit.script function.

Reply:
But we cannot index the ModuleList containing the poolers in a script_method, apparently, so we don't get to have one loop.

Reply:
I think having two loops for now is fine, that shouldn't affect the runtime.

Reply:
OK